crates/uv-resolver/src/resolver/index.rs

```diff
@@ -7,34 +7,27 @@ use uv_normalize::PackageName;
 use crate::resolver::provider::{MetadataResponse, VersionsResponse};
 
 /// In-memory index of package metadata.
+#[derive(Default, Clone)]
+pub struct InMemoryIndex(Arc<SharedInMemoryIndex>);
+
 #[derive(Default)]
-pub struct InMemoryIndex {
+struct SharedInMemoryIndex {
     /// A map from package name to the metadata for that package and the index where the metadata
     /// came from.
-    pub(crate) packages: OnceMap<PackageName, Rc<VersionsResponse>>,
+    pub(crate) packages: OnceMap<PackageName, Arc<VersionsResponse>>,
 
     /// A map from package ID to metadata for that distribution.
-    pub(crate) distributions: OnceMap<VersionId, Rc<MetadataResponse>>,
+    pub(crate) distributions: OnceMap<VersionId, Arc<MetadataResponse>>,
```

Comment:
Since the outer type isn't `pub(crate)`, should the fields be? I think it should be okay to un-export them.